bigframes/dataframe.py

```diff
@@ -663,22 +664,24 @@ def _apply_binop(
         how: str = "outer",
         reverse: bool = False,
     ):
-        if isinstance(other, (float, int)):
+        if isinstance(other, (float, int, bool)):
             return self._apply_scalar_binop(other, op, reverse=reverse)
-        elif isinstance(other, indexes.Index):
-            return self._apply_series_binop(
-                other.to_series(index=self.index),
-                op,
-                axis=axis,
-                how=how,
-                reverse=reverse,
-            )
-        elif isinstance(other, bigframes.series.Series):
-            return self._apply_series_binop(
-                other, op, axis=axis, how=how, reverse=reverse
-            )
         elif isinstance(other, DataFrame):
             return self._apply_dataframe_binop(other, op, how=how, reverse=reverse)
+        elif isinstance(other, pandas.DataFrame):
```

Comment:
might need to add pandas.DataFrame to the type hint for other?

Reply:
For what purpose? Mypy is able to infer based on the `isinstance` checks.

Reply:
Oh, gotcha, but should we remove the other type hints then? Just for consistency/cleanliness.